addons/l10n_pk_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,454 @@
+            )
+            if (sale_group_id := self.env['ir.model.data']._xmlid_to_res_model_res_id(
+                f"account.{company_id.id}_{ref}"
+            )[1])
+        ])
+
+    def _get_l10n_pk_withhold_groups(self, company_id):
+        return self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_wt')
+
+    def _get_move_applicability(self, move):
+        # EXTENDS account_edi
+        self.ensure_one()
+        if self.code != 'pk_einvoice_1_3':
+            return super()._get_move_applicability(move)
+        is_taxable = self._get_l10n_pk_sales_groups(move.company_id) & move.line_ids.tax_ids.tax_group_id
+        if (
+            (move.is_sale_document() or move.is_purchase_document())
+            and move.country_code == 'PK'
+            and is_taxable
+        ):
+            return {
+                'post': self._l10n_pk_edi_post_invoice,
+                'cancel': self._l10n_pk_edi_cancel_invoice,
+                'edi_content': json.dumps(
+                    self._l10n_pk_edi_generate_invoice_json(move)
+                ).encode()
+            }
+
+    def _check_move_configuration(self, move):
+        if self.code != "pk_einvoice_1_3":
+            return super()._check_move_configuration(move)
+        error_messages = []
+        for line in move.invoice_line_ids.filtered(lambda line: line.display_type == 'product'
+            and not self._l10n_pk_is_global_discount(line)):
+            if line.discount < 0:
+                error_messages.append(_("Negative discount is not allowed, set in line %s", line.name))
+            if not line.product_id.l10n_pk_edi_sale_type_id:
+                error_messages.append(_("Sale type is not defined for line %s", line.name))
+            if not line.product_id.l10n_pk_edi_schedule_code_id:
+                error_messages.append(_("Schedule code is not defined for line %s", line.name))
+            hs_code = self._l10n_pk_edi_extract_digits(line.product_id.hs_code)
+            if not hs_code:
+                error_messages.append(_("HS code is not set in product line %s", line.name))
+            elif not re.match("^[a-zA-Z0-9]{8}$", hs_code):
+                error_messages.append(_(
+                    "Invalid HS Code (%(hs_code)s) in product line %(product_line)s",
+                        hs_code=hs_code,
+                        product_line=line.product_id.name or line.name
+                    )
+                )
+        return error_messages
+
+    def _needs_web_services(self):
+        self.ensure_one()
+        return self.code == "pk_einvoice_1_3" or super()._needs_web_services()
+
+    def _l10n_pk_validate_seller(self, partner):
+        self.ensure_one()
+        messages = []
+        if not re.match("^.{3,100}$", partner.street or ""):
+            messages.append(_("- Street required min 3 and max 100 characters"))
+        if not partner.street2 and not re.match("^.{3,100}$", partner.street2):
```

Comment:
```suggestion
        if not re.match("^.{3,100}$", partner.street2 or ""):
```
Better to be consistent. With above and below `if` blocks